### DISCUSSION TITLE
Fianance Chart - Custom Floating X & Y Axis Labels

### DISCUSSION THREAD

**Discussion by jasonsullivan** [Live-Charts/LiveCharts2#883] [General]
Hello

I am looking for some advice and/or guidence on adding floating axis labels to a Finance Chart.

In the attached screenshot from another charting platform, I have included an example of what this looks like.

The blue label moves up and down the Y Axis and matches the last close price value related to the bar base on a realtime data feed.

The orange labels movement tracks the position of the mouse.  I have only shown the Y Axis in the screenshot, but there is a label that does the same on the X Axis.

![image](https://user-images.githubusercontent.com/6076840/211296787-772fb594-a46c-4248-9d6e-a3c0143d7f0e.png)

Is there a built-in method currently available in LiveCharts2 that would allow me to acheive a similar behaviour?

Can I roll my own based on capabilities already in the underlying LiveCharts2 platform and would you be able to provide an example of this approach?

Kind regards
Jason

**Comment by jasonsullivan** [comment]
Couldn't see the wood for the trees ;-)

![image](https://user-images.githubusercontent.com/6076840/211300538-1943f69d-2f23-4bd2-957b-2211cc8af770.png)

**Reply by EthanBlank-Git** [reply]
How were you able to get the crosshair to only show the axis labels? (not the lines themselves) and track with the latest value on the chart? This was also the inspiration for [my discussion post](https://github.com/beto-rodriguez/LiveCharts2/discussions/546#discussion-4203337)
Also, do you have any issues with a 'live' collection of candlesticks/financial points? My program crashes from (I assume) trying to access the data of the collection while it is constantly changing

**Reply by jasonsullivan** [reply]
Hi @EBTheKidd, thanks for responding ;-) 

In my screenshot in my original post, I am showing a chart from FXCM's Tradestation to illustrate the goal, but I was able to 'hide' the crosshair lines by setting the CrossHairPaint property Color to Transparent and its StrokeThickness to 0.

![image](https://user-images.githubusercontent.com/6076840/211948964-bc0c309e-6289-473f-bd74-057538027016.png)

I have not worked out how to animate it so that it tracks the last price yet though - I'm not sure it can be used for this purpose either.  

This led me to experimenting with Syncfusions SFChart.  Although I havent plugged in live data yet, it was a trivial exercise to add a HorizontalLineAnnotation and Y axis binding that would take care of the animation once it is wired up.

`            <syncfusionChart:SfChart.Annotations>
                <syncfusionChart:HorizontalLineAnnotation
                    x:Name="annotation"
                    ShowAxisLabel="True"
                    ShowLine="True"
                    Y1="{Binding PriceAnnotation}"></syncfusionChart:HorizontalLineAnnotation>
            </syncfusionChart:SfChart.Annotations>`

I am using the CommunityToolkit.MVVM and it auto generates partial methods.

`        /// <inheritdoc cref="_priceAnnotation"/>
        [global::System.CodeDom.Compiler.GeneratedCode("CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator", "8.0.0.0")]
        [global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
        public double PriceAnnotation
        {
            get => _priceAnnotation;
            set
            {
                if (!global::System.Collections.Generic.EqualityComparer<double>.Default.Equals(_priceAnnotation, value))
                {
                    OnPriceAnnotationChanging(value);
                    OnPropertyChanging(global::CommunityToolkit.Mvvm.ComponentModel.__Internals.__KnownINotifyPropertyChangingArgs.PriceAnnotation);
                    _priceAnnotation = value;
                    OnPriceAnnotationChanged(value);
                    OnPropertyChanged(global::CommunityToolkit.Mvvm.ComponentModel.__Internals.__KnownINotifyPropertyChangedArgs.PriceAnnotation);
                }
            }
        }`

 `        /// <summary>Executes the logic for when <see cref="PriceAnnotation"/> is changing.</summary>
        [global::System.CodeDom.Compiler.GeneratedCode("CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator", "8.0.0.0")]
        partial void OnPriceAnnotationChanging(double value);
        /// <summary>Executes the logic for when <see cref="PriceAnnotation"/> just changed.</summary>
        [global::System.CodeDom.Compiler.GeneratedCode("CommunityToolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator", "8.0.0.0")]
        partial void OnPriceAnnotationChanged(double value);`        

I get the last value and assign it to PriceAnnotation once the collection bound to my chart has changed:
`
        private async Task GetInstrumentPeriodHistoryAsync()
        {
            var request = new InstrumentRequestEventArgs(SelectedInstrument!.Name!, SelectedInstrumentPeriod!.Interval, SelectedInstrumentPeriod.IntervalCompression);
            SelectedInstrumentPeriodPrices = await _slackcatsData.GetInstrumentPriceHistoriesAsync(request);
        }
        
partial void OnSelectedInstrumentPeriodPricesChanged(ObservableCollection<InstrumentPriceHistoryModel>? value)
        {
            if (value is not null)
            {
                SelectedInstrumentPeriod!.InstrumentPriceHistory = value;
                PriceAnnotation = value.OrderBy(o=>o.TimeStamp).Last().AskClose;
            }
        }`
Hope this is helpful to you.

**Reply by jasonsullivan** [reply]
![image](https://user-images.githubusercontent.com/6076840/211954245-14f53a5d-f9c2-4d0d-807b-ad561ad31fcf.png)

**Reply by EthanBlank-Git** [reply]
Interesting, thank you for taking the time to write a response!! I would imagine that some point in the future we will be able to specify which value is being used to display the crosshair (in the case of a financial point: open, high, low, or close). Fingers crossed!
